libgdf/src/transpose.cu

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2018 NVIDIA Corporation.  All rights reserved. */
+
+#include "gdf_table.cuh"
+#include <gdf/gdf.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <memory>
+#include <stdio.h>
+
+gdf_error gdf_transpose(size_t ncols,
+                        gdf_column** in_cols,
+                        gdf_column** out_cols)
+{
+    // Make sure the inputs are not null
+    if( (0 == ncols) 
+        || (nullptr == in_cols) 
+        || (nullptr == out_cols))
+    {
+        return GDF_DATASET_EMPTY;
+    }
+
+    // If there are no rows in the input, return successfully
+    if ( 0 == in_cols[0]->size )
+    {
+        return GDF_SUCCESS;
+    }
+
```

Comment:
@kkraus14 @randerzander assuming we're going to be okay with the restriction that all the columns have to be the same type for a transpose, then you should add a check to ensure that all of the columns have the same type (both input and output columns).

Reply:
Actually I need to find out whether all input columns need to be homogeneous or can they be simply numerical. In which case, the `get_packed_row_values` method would need an update.

Reply:
If non-homogenous but numerical we'd need to do a type promotion to avoid mixing int8 with int32 for example.